src/server/scripts/Kalimdor/HallsOfOrigination/instance_halls_of_origination.cpp

```diff
@@ -207,9 +259,15 @@ class instance_halls_of_origination : public InstanceMapScript
 
             void ReadSaveDataMore(std::istringstream& data) override
             {
-                uint32 deadElementals;
-                data >> deadElementals;
-                IncreaseDeadElementals(deadElementals);
+                data >> _deadElementals;
+
+                if (_deadElementals == 4)
+                {
+                    if (GameObject* mirror = instance->GetGameObject(SunMirrorGUID))
+                        mirror->SetGoState(GO_STATE_ACTIVE);
+                    if (GameObject* door = instance->GetGameObject(AnraphetDoorGUID))
+                        door->SetGoState(GO_STATE_ACTIVE);
```

Comment:
i don't think any gameobject is spawned at this moment

Reply:
It's just rewritten, should it be moved to OnGameObjectCreate()?

Reply:
should be better